aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -93,146 +165,126 @@ module AptosFramework::Token {
 
     /// Represent collection and token metadata for a creator
     struct Collections has key {
-        collections: Table<ASCII::String, Collection>,
-        token_data: Table<TokenId, TokenData>,
+        collections: Table<String, Collection>,
+        token_data: Table<TokenDataId, TokenData>,
         burn_capabilities: Table<TokenId, BurnCapability>,
         mint_capabilities: Table<TokenId, MintCapability>,
         create_collection_events: EventHandle<CreateCollectionEvent>,
         create_token_events: EventHandle<CreateTokenEvent>,
-        mint_token_events: EventHandle<MintTokenEvent>,
+        mint_token_events: EventHandle<MintTokenEvent>
     }
 
     /// Represent the collection metadata
     struct Collection has store {
         // Describes the collection
-        description: ASCII::String,
+        description: String,
         // Unique name within this creators account for this collection
-        name: ASCII::String,
+        name: String,
         // URL for additional information /media
-        uri: ASCII::String,
-        // Total number of distinct Tokens tracked by the collection
+        uri: String,
+        // Total number of distinct TokenData tracked by the collection
         count: u64,
-        // Optional maximum number of tokens allowed within this collections
-        maximum: Option<u64>,
```

Comment:
probably good to make this still optional, that way there's not something arbitrary here.

Reply:
as explained in other places. The ergonomic API can be provided in SDK